google_guest_agent/metadata_integ_test.go

```diff
@@ -0,0 +1,30 @@
+// +build integration
+
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+const (
+	malformedKey  = "malformed-ssh-keys"
+	malformedKey2 = ":malformed-ssh-keys"
+)
+
+func TestMalformedSSHKeys(t *testing.T) {
+	// Insert a malformed ssh key
+	newMetadata.Instance.Attributes.SSHKeys = append(newMetadata.Instance.Attributes.SSHKeys, malformedKey)
+	time.Sleep(5 * time.Second)
+
+	if exist, err := userExists(malformedKey); exist {
+		t.Fatalf("%s user should not exist", malformedKey)
```

Comment:
i don't understand this, can you explain it? we add a key to a global variable, and then we sleep. who is going to act on that global? i think this test will always succeed. maybe add a well-formed key as well and check a user DOES exist, to validate the test logic. it can be a separate test function

Reply:
I just aware that the test is not the same process as guest-agent daemon. so probably, we need to think a way to force act on that global?

Reply:
right, by invoking some function. we can't get the guest agent to do anything, it is some other running process.

Reply:
Only implement unit test for this. Resolve this.